ClearKeep/App/System/PushNotificationsHandler.swift

```diff
@@ -31,23 +35,123 @@ extension PushNotificationsHandler: UNUserNotificationCenterDelegate {
 								willPresent notification: UNNotification,
 								withCompletionHandler completionHandler:
 								@escaping (UNNotificationPresentationOptions) -> Void) {
+		
+		self.handleNotification(userInfo: notification.request.content.userInfo, completionHandler: {})
 		completionHandler([.banner, .sound, .list])
 	}
 	
 	func userNotificationCenter(_ center: UNUserNotificationCenter,
 								didReceive response: UNNotificationResponse,
 								withCompletionHandler completionHandler: @escaping () -> Void) {
 		let userInfo = response.notification.request.content.userInfo
-		handleNotification(userInfo: userInfo, completionHandler: completionHandler)
+		print("notification payload: \(userInfo)")
+		self.handleNotification(userInfo: userInfo, completionHandler: completionHandler)
 	}
 	
 	func handleNotification(userInfo: [AnyHashable: Any], completionHandler: @escaping () -> Void) {
-		guard let payload = userInfo["aps"] as? NotificationPayload else {
-			completionHandler()
+		if let jsonString = userInfo["publication"] as? String,
+		   let jsonData = jsonString.data(using: .utf8) {
+			
+			if let publicationJson = try? JSONSerialization.jsonObject(with: jsonData, options: []) as? [String: Any],
+			   let deactiveAccountId = publicationJson["deactive_account_id"] as? String {
+				handleLogout(deactiveAccountId: deactiveAccountId)
+				completionHandler()
+				return
+			}
+			if let publicationRemove = try? JSONSerialization.jsonObject(with: jsonData, options: []) as? [String: Any],
+			   let remove = publicationRemove["leave_member"] as? String {
+				handleAlert()
+				return
+			}
+
+			if let publicationRemove = try? JSONSerialization.jsonObject(with: jsonData, options: []) as? [String: Any],
+			   let add = publicationRemove["added_member_id"] as? String {
+				DispatchQueue.main.asyncAfter(deadline: .now() + 0.05, execute: {
+					NotificationCenter.default.post(name: NSNotification.reloadDataHome, object: nil)
+				})
+				return
+			}
+
+			do {
+				let publication = try JSONDecoder().decode(PublicationMessageNotification.self, from: jsonData)
+				//container.appState[\.authentication.selectedGroupId] = publication.groupId
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/8649f3f2f0ce9de1db3f8c96a6cf34266ff6e2d9/ClearKeep/App/System/PushNotificationsHandler.swift#L77"></span>Prefer at least one space after slashes for comments.<br />
<code>comment_spacing</code> <code>PushNotificationsHandler.swift:77</code></td>
    </tr>
  </tbody>
</table>